Src/FluentAssertions/Collections/StringCollectionAssertions.cs

```diff
@@ -190,5 +191,50 @@ public AndConstraint<StringCollectionAssertions> Contain(IEnumerable<string> exp
             args.AddRange(becauseArgs);
             return base.NotContain(unexpected, because, args.ToArray());
         }
+
+        /// <summary>
+        /// Asserts that the collection contains at least one string that matches a wildcard pattern.
+        /// </summary>
+        /// <param name="wildcardPattern">
+        /// The wildcard pattern with which the subject is matched, where * and ? have special meanings.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<StringCollectionAssertions> ContainMatch(string wildcardPattern, string because = "",
```

Comment:
❓ Should we instead of returning an `AndConstraint` return an `AndWhichConstraint`, such that we include the matched `item`?
Downside: If there are multiple matches, we would only return the first match.

Reply:
That's a good point. 

Another possible alternative is to return all matches.

The downside to that is that the whole string enumerable will have to be iterated, instead of returning at the first match.

Reply:
Aha! 
`SelfReferencingCollectionAssertions.Contain(Expression<Func<T, bool>> predicate` returns an `AndWhichConstraint`.
https://github.com/fluentassertions/fluentassertions/blob/dbc0d5f6e301f05b2b096e2bb29414c8cf66c546/Src/FluentAssertions/Collections/SelfReferencingCollectionAssertions.cs#L461-L466

If invoking `.Which` on the returned `AndWhichConstraint` it *then* checks if there was more than one result.

Reply:
Should I provide an example in the docs about `.Which` if this is the accepted API?

Reply:
I'm sorry, I forgot to include the complete desired signature.
My thought was that the return type should be `AndWhichConstraint<StringCollectionAssertions, string>`. 
From what I could see, that's whats the most places in FA uses.

Reply:
So return first match only?

Reply:
Yes.
That seems most consistent with the API as that's what all other assertions use, at least as far as I could see.
Any input @dennisdoomen?

So 
```c#
return new AndWhichConstraint<StringCollectionAssertions, string>(this, AllThatMatch(wildcardPattern));
```

Reply:
If that is the case, do you prefer implementation A or B:

A)
```csharp
string match;
using (var scope = new AssertionScope())
{
    match = Subject?.FirstOrDefault(item =>
    {
        item.Should().Match(wildcardPattern);
        return !scope.Discard().Any();
    });
}

Execute.Assertion
    .BecauseOf(because, becauseArgs)
    .ForCondition(Subject is object)
    .FailWith("Expected {context:collection} to contain a match of {0}{reason}, but found <null>.", wildcardPattern)
    .Then
    .ForCondition(match is object)
    .FailWith("Expected {context:collection} {0} to contain a match of {1}{reason}.", Subject, wildcardPattern);

return new AndWhichConstraint<StringCollectionAssertions, string>(this, match);
```

B)
```csharp
public AndWhichConstraint<StringCollectionAssertions, string> ContainMatch(string wildcardPattern, string because = "",
    params object[] becauseArgs)
{
    Execute.Assertion
        .BecauseOf(because, becauseArgs)
        .ForCondition(Subject is object)
        .FailWith("Expected {context:collection} to contain a match of {0}{reason}, but found <null>.", wildcardPattern)
        .Then
        .ForCondition(ContainsMatch(wildcardPattern))
        .FailWith("Expected {context:collection} {0} to contain a match of {1}{reason}.", Subject, wildcardPattern);

    return new AndWhichConstraint<StringCollectionAssertions, string>(this, FirstMatch(wildcardPattern));
}

private bool ContainsMatch(string wildcardPattern)
{
    using (var scope = new AssertionScope())
    {
        return Subject.Any(item =>
        {
            item.Should().Match(wildcardPattern);
            return !scope.Discard().Any();
        });
    }
}

private string FirstMatch(string wildcardPattern)
{
    return Subject.FirstOrDefault(item =>
    {
        using (var scope = new AssertionScope())
        {
            item.Should().Match(wildcardPattern);
            return !scope.Discard().Any();
        }
    });
}
```

Reply:
@jnyrup Oh, I just realized if the `AndWhichConstraint` is passed an enumerable with more than 1 element, when accessing `Which` FA throws an exception stating `More than one object found. FluentAssertions cannot determine which object is meant.` Is this the desired behavior you're after?

Reply:
Yes.
Again sorry for the confusion, I somehow thought you had seen that.

Reply:
Haha, no worries, it's part of learning the framework.

Since we were talking about 'first match only' I had assumed first match, discard other matches. Did not consider throwing an exception, but it seems like this is the default behavior for `.Which`